### ISSUE TITLE
Idea for syntax based on decorators

### ISSUE THREAD

**Issue by littledan** [rbuckton/proposal-refs#11]
What if the syntax for `ref` were instead, decorators applied to variables or member expressions? I think this would be in parallel to the semantics for field decorators, and it would allow for more flexibility. For example

```js
@a x
@b x.y
```
would desugar into
```js
a({get: () => x, set: (v) => x = v}, {type: "ref"})
b({get: () => x.y, set: (v) => x.y = v}, {type: "ref"})
```

One benefit here would be that you can use different decorators to do different things. For example, one big use case in the explainer was to make a deferred, read-only reference to something which is currently in tdz and will be filled in later. This could be done in a "good-enough" way by returning a Proxy which initially throws while the `get` throws a TDZ error, and later, defers to its target. (Proxy invariants mean that it won't be able to reflect the change if a frozen object is overwritten with a non-frozen object, but this need does not come up for the use cases in the explainer.)

Another benefit is that the syntax is "simpler" in that there is no new contextual keyword or `[no LineTerminator here]` exception.

What do you think?
